ui/hooks/useUserPreferencedCurrency.js

```diff
@@ -34,9 +36,10 @@ import { useMultichainSelector } from './useMultichainSelector';
  * useUserPreferencedCurrency
  *
  * returns an object that contains what currency to use for displaying values based
- * on the user's preference settings, as well as the significant number of decimals
+ * on whether the user needs to check showNativeTokenAsMainBalance setting, as well as the significant number of decimals
```

Comment:
How are you thinking about the purpose of this hook (and the other `user-preferred-currency...` components) now that the fiat/crypto preference setting is removed?  

Since `showNativeTokenAsMainBalance` only affects the balance on top of the home page, it seems like all other the places using `user-preferred...` would no longer need to care about that primary/secondary distinction.

Are they still there because it keeps the change smaller?  Do you think there's an opportunity to simplify later?

Reply:
Its true that it is keeping the change smaller, but also this hook is still handling fallbacks and cases where `showFiat` is false; if for instance we are on a testnet and user do not have the setting "show conversion on testnet" we are returning "native return"; so even if we refactor this in future; i still think its good to have these checks in a separate hook;

Reply:
Makes sense we can keep for now.  Probably some room to clean up in the future.  It's a little messy for these generic hooks to have logic specific to 1 caller.   Could be nicer to keep that logic closer to the main balance component itself.